python/cudf/cudf/core/column/numerical.py

```diff
@@ -578,19 +591,34 @@ def can_cast_safely(self, to_dtype: DtypeObj) -> bool:
         Returns true if all the values in self can be
         safely cast to dtype
         """
-        if self.dtype.kind == to_dtype.kind:
-            if self.dtype <= to_dtype:
+        # Convert potential pandas extension dtypes to numpy dtypes
+        # For example, convert Int32Dtype to np.dtype('int32')
+        self_dtype_numpy = (
+            np.dtype(self.dtype.numpy_dtype)
+            if hasattr(self.dtype, "numpy_dtype")
+            else self.dtype
+        )
+        to_dtype_numpy = (
+            np.dtype(to_dtype.numpy_dtype)
+            if hasattr(to_dtype, "numpy_dtype")
+            else to_dtype
+        )
+
+        if self_dtype_numpy.kind == to_dtype_numpy.kind:
+            # Check if self dtype can be safely cast to to_dtype
+            # For same kinds, we can compare the sizes
+            if self_dtype_numpy <= to_dtype_numpy:
                 return True
             else:
                 # Kinds are the same but to_dtype is smaller
-                if "float" in to_dtype.name:
-                    finfo = np.finfo(to_dtype)
+                if "float" in to_dtype_numpy.name:
+                    finfo = np.finfo(to_dtype_numpy)
                     lower_, upper_ = finfo.min, finfo.max
-                elif "int" in to_dtype.name:
-                    iinfo = np.iinfo(to_dtype)
+                elif "int" in to_dtype_numpy.name:
+                    iinfo = np.iinfo(to_dtype_numpy)
                     lower_, upper_ = iinfo.min, iinfo.max
```

Comment:
Can we move this after the np.isnan check? It doesn't look like we need this information in that case because we return early.

Reply:
Done.